hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/volume/HddsVolume.java

```diff
@@ -343,35 +357,41 @@ public void loadDbStore(boolean readOnly) throws IOException {
     File clusterIdDir = new File(dbVolume == null ?
         getStorageDir() : dbVolume.getStorageDir(),
         getClusterID());
-    if (!clusterIdDir.exists()) {
-      throw new IOException("Working dir " + clusterIdDir.getAbsolutePath() +
-          " not created for HddsVolume: " + getStorageDir().getAbsolutePath());
-    }
+    try {
+      if (!clusterIdDir.exists()) {
+        throw new IOException("Working dir " + clusterIdDir.getAbsolutePath() +
+            " not created for HddsVolume: " + getStorageDir().getAbsolutePath());
+      }
 
-    File storageIdDir = new File(clusterIdDir, getStorageID());
-    if (!storageIdDir.exists()) {
-      throw new IOException("Db parent dir " + storageIdDir.getAbsolutePath() +
-          " not found for HddsVolume: " + getStorageDir().getAbsolutePath());
-    }
+      File storageIdDir = new File(clusterIdDir, getStorageID());
+      if (!storageIdDir.exists()) {
+        throw new IOException("Db parent dir " + storageIdDir.getAbsolutePath() +
+            " not found for HddsVolume: " + getStorageDir().getAbsolutePath());
+      }
 
-    File containerDBFile = new File(storageIdDir, CONTAINER_DB_NAME);
-    if (!containerDBFile.exists()) {
-      throw new IOException("Db dir " + storageIdDir.getAbsolutePath() +
-          " not found for HddsVolume: " + getStorageDir().getAbsolutePath());
-    }
+      File containerDBFile = new File(storageIdDir, CONTAINER_DB_NAME);
+      if (!containerDBFile.exists()) {
+        throw new IOException("Db dir " + storageIdDir.getAbsolutePath() +
+            " not found for HddsVolume: " + getStorageDir().getAbsolutePath());
+      }
 
-    String containerDBPath = containerDBFile.getAbsolutePath();
-    try {
-      initPerDiskDBStore(containerDBPath, getConf(), readOnly);
+      String containerDBPath = containerDBFile.getAbsolutePath();
+      try {
+        initPerDiskDBStore(containerDBPath, getConf(), readOnly);
+      } catch (IOException e) {
+        throw new IOException("Can't init db instance under path "
+            + containerDBPath + " for volume " + getStorageID(), e);
+      }
+
+      dbParentDir = storageIdDir;
+      dbLoaded.set(true);
+      dbLoadFailure.set(false);
+      LOG.info("SchemaV3 db is loaded at {} for volume {}", containerDBPath,
+          getStorageID());
     } catch (IOException e) {
-      throw new IOException("Can't init db instance under path "
-          + containerDBPath + " for volume " + getStorageID(), e);
+      dbLoadFailure.set(true);
```

Comment:
@xichen01 , thanks for reporting this.  Can you share more detail about which operation will cause the DB load failure and throw the IOException, besides the known initPerDiskDBStore?  Is there any exception stack can be shared?

Reply:
Not sure of the root cause of the DB loading issue, but we found a very large `container.db` in the system home directory, as well as error logs from `HddsVolume` initing.


logs from `HddsVolume` initing
![image](https://github.com/apache/ozone/assets/32928346/be468fd7-58e0-4925-b3f1-9305506b11b9)

`container.db` in the system home
![image](https://github.com/apache/ozone/assets/32928346/add77760-9a32-4e13-9ee4-3deb44a3766e)

Reply:
1. Manually destroy the RocksDB file. 2. Then restart DN and write data.  can reproduce a similar problem.

Reply:
```
 ERROR org.apache.hadoop.ozone.container.ozoneimpl.OzoneContainer: Load db store for HddsVolume /xxx/xxx/xxx/ozone/hdds failed
java.io.IOException: Can't init db instance under path /xxx/xxxx/xxx/ozone/hdds/CID-xxx/DS-xxx/container.db for volume DS-xxx
        at org.apache.hadoop.ozone.container.common.volume.HddsVolume.loadDbStore(HddsVolume.java:235)
        at org.apache.hadoop.ozone.container.common.utils.HddsVolumeUtil.loadAllHddsVolumeDbStore(HddsVolumeUtil.java:99)
        at org.apache.hadoop.ozone.container.ozoneimpl.OzoneContainer.<init>(OzoneContainer.java:146)
        at org.apache.hadoop.ozone.container.common.statemachine.DatanodeStateMachine.<init>(DatanodeStateMachine.java:153)
        at org.apache.hadoop.ozone.HddsDatanodeService.start(HddsDatanodeService.java:295)
        at org.apache.hadoop.ozone.HddsDatanodeService.start(HddsDatanodeService.java:227)
        at org.apache.hadoop.ozone.HddsDatanodeService.call(HddsDatanodeService.java:195)
        at org.apache.hadoop.ozone.HddsDatanodeService.call(HddsDatanodeService.java:104)
        at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
        at picocli.CommandLine.access$1300(CommandLine.java:145)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2352)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2346)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2311)
        at picocli.CommandLine$AbstractParseResultHandler.handleParseResult(CommandLine.java:2172)
        at picocli.CommandLine.parseWithHandlers(CommandLine.java:2550)
        at picocli.CommandLine.parseWithHandler(CommandLine.java:2485)
        at org.apache.hadoop.hdds.cli.GenericCli.execute(GenericCli.java:96)
        at org.apache.hadoop.hdds.cli.GenericCli.run(GenericCli.java:87)
        at org.apache.hadoop.ozone.HddsDatanodeService.main(HddsDatanodeService.java:178)
Caused by: java.io.IOException: Failed init RocksDB, db path : /xxx/xxx/xxx/ozone/hdds/CID-xxx/DS-xxx/container.db, exception :org.rocksdb.RocksDBException CURRENT file does not end with newline
        at org.apache.hadoop.hdds.utils.db.RDBStore.<init>(RDBStore.java:130)
        at org.apache.hadoop.hdds.utils.db.DBStoreBuilder.build(DBStoreBuilder.java:191)
        at org.apache.hadoop.ozone.container.metadata.AbstractDatanodeStore.start(AbstractDatanodeStore.java:146)
        at org.apache.hadoop.ozone.container.metadata.AbstractDatanodeStore.<init>(AbstractDatanodeStore.java:99)
        at org.apache.hadoop.ozone.container.metadata.DatanodeStoreSchemaThreeImpl.<init>(DatanodeStoreSchemaThreeImpl.java:54)
        at org.apache.hadoop.ozone.container.keyvalue.helpers.BlockUtils.getUncachedDatanodeStore(BlockUtils.java:84)
        at org.apache.hadoop.ozone.container.common.utils.HddsVolumeUtil.initPerDiskDBStore(HddsVolumeUtil.java:79)
        at org.apache.hadoop.ozone.container.common.volume.HddsVolume.loadDbStore(HddsVolume.java:232)
        ... 18 more
```

Reply:
I think we need to root cause this issue. Could you kindly check the related DN logs to see if there is any related suspicious logs?  If we don't have the root cause, then this extra IOException catch may not enough.

Reply:
Not found others related log which can help to locate the root cause, this may be a legacy issue that may have existed when the cluster was created, but with this PR, we could have found this issue faster.

Reply:
I have observed another impact when volume metadata path is `removed incorrectly in test env`. But at that point of time, container was getting loaded having NPE in other flow.
NPE was handled with https://github.com/apache/ozone/pull/5921 avoiding loading of container.
Since it was test script issue where cleanup was not proper, referring similar case.

Reply:
@xichen01 , in case we don't know the exact case of DB loading failure, let's change this catch from IOException to Throwable.

Reply:
@xichen01 , could you please address my above comment and provide a new patch?

Reply:
OK, I will